src/Engine/ProtoImperative/CodeGen.cs

```diff
@@ -2634,6 +2654,21 @@ private void EmitForLoopNode(ImperativeNode node, ref ProtoCore.Type inferredTyp
 
                 type.UID = (int)ProtoCore.PrimitiveType.kTypeVoid;
                 EmitWhileStmtNode(whileStatement, ref type, isBooleanOp, graphNode);
+
+                // As we add a dummy code block around forloop node, RETCN 
+                // instruction will get debugging information from 
+                // localCodeBlockNode, which is forloop node. We temporarily set
+                // lcoalCodeBlockNode to a dummy node so that RETCN won't have 
+                // debugging information.
+                var dummyCodeBlockNode = new CodeBlockNode();
+                var backUpLocalCodeBlockNode = localCodeBlockNode;
+                localCodeBlockNode = dummyCodeBlockNode;
+
+                EmitInstrConsole(ProtoCore.DSASM.kw.retcn);
+                EmitRetcn(localCodeBlock.codeBlockId);
```

Comment:
Is it necessary to create dummyCodeBlockNode and backUpLocalCodeBlockNode? 

Because it seems the only thing needed here is the codeblock id in EmitRetcn(localCodeBlock.codeBlockId);